server/commands/battle.py

```diff
@@ -0,0 +1,597 @@
+            emoji = "🔎"
+        else:
+            emoji = "⚔️"
+        msg += (
+            f"{emoji} [{client.id}]{client.battle.fighter} ({client.showname}) {emoji}"
+        )
+    return msg
+
+
+def ooc_cmd_battle_info(client, arg):
+    """
+    Send you info about the battle.
+    Usage: /battle_info
+    """
+    if client in client.area.fighters:
+        msg = send_battle_info(client)
+        client.send_ooc(msg)
+    else:
+        client.send_ooc("You are not fighting!")
+
+
+def ooc_cmd_fight(client, arg):
+    """
+    Allow you to join the battle!
+    Usage: /fight
+    """
+    if client.area.can_battle:
+        if client.battle is not None:
+            if client not in client.area.fighters:
+                if not client.area.battle_started:
+                    client.area.fighters.append(client)
+                    msg = send_battle_info(client)
+                    for client in client.area.fighters:
+                        client.send_ooc(msg)
+                    client.area.broadcast_ooc(
+                        f"⚔️{client.battle.fighter} ({client.showname}) is ready to fight!⚔️"
+                    )
+                    fighter_name = client.area.area_manager.char_list[client.char_id]
+                    client.area.send_ic(
+                        client=None,
+                        msg_type="1",
+                        pre=client.last_sprite,
+                        folder=fighter_name,
+                        anim=client.last_sprite,
+                        msg=f"~{client.battle.fighter}~ is ready to fight",
+                        pos=client.pos,
+                        sfx="",
+                        emote_mod=0,
+                        cid=-1,
+                        sfx_delay=0,
+                        button=0,
+                        evidence=[0],
+                        flip=client.flip,
+                        ding=0,
+                        color=3,
+                        showname="",
+                        charid_pair=client.charid_pair,
+                        other_folder="",
+                        other_emote="",
+                        offset_pair=0,
+                        other_offset=0,
+                        other_flip=0,
+                        nonint_pre=0,
+                        sfx_looping="0",
+                        screenshake=0,
+                        frames_shake="",
+                        frames_realization="",
+                        frames_sfx="",
+                        additive=0,
+                        effect="",
+                        targets=None,
+                    )
+                else:
+                    client.send_ooc("The battle is already started!")
+            else:
+                client.send_ooc("You are already in battle!")
+        else:
+            client.send_ooc("You have to choose a fighter to start a battle!")
+    else:
+        client.send_ooc("You cannot fight in this area!")
```

Comment:
if if if if else else else else staircase is very hard to read, please revise

Reply:
Done!